gax/src/main/java/com/google/api/gax/tracing/TracedBatchingCallable.java

```diff
@@ -67,7 +67,8 @@ public TracedBatchingCallable(
   public ApiFuture<ResponseT> futureCall(RequestT request, ApiCallContext context) {
     // NOTE: This will be invoked asynchronously outside of the original caller's thread.
     // So this start a top level tracer.
-    ApiTracer tracer = tracerFactory.newRootTracer(spanName);
+    ApiTracer tracer =
```

Comment:
Should stuff like this happen inside the `try` block (here and in all similar places)? I'm worrying about potential `NullPointerException` if `context` is `null`. Before this call it would throw `NPE` on L80, which will be caught on L85 and reported by the `finisher` (it won't happen with this change).

Reply:
ApiCallContext should never be null, the defaultCallContext callable will ensure that. I think moving the tracer creation to the try block will create noise: I would still need to keep the variable declaration for both the tracer & the finisher outside the try and then do a null check in the catch